### DISCUSSION TITLE
Background task not working after deploy on server

### DISCUSSION THREAD

**Discussion by saleemhaq** [OrchardCMS/OrchardCore#10938] [Q&A]
My background task is working locally but when i deployed on server it not working. The job responsibility is to publish and unpublish content item. 

The implementation is below:

  foreach (var item in publishPostsPromotionDetails)
            {
                    item.Published = false;
                    
                    await _contentManager.UpdateAsync(item);
                }
            }

Please help.

**Comment by jtkech** [comment]
Assuming that you defined an `IBackgroundTask`.

When the app is running, it needs at least one request to build the tenant / shell and then start the bg task.

There is a pending a PR #10638 that will allow to warmup the tenant / shell to not have to wait for a first request.

That said, on a prod server if your app is recycled the bg task is stopped, here you may need a way to keep it alive.

**Reply by saleemhaq** [reply]
> Assuming that you defined an `IBackgroundTask`.
> 
> When the app is running, it needs at least one request to build the tenant / shell and then start the bg task.
> 
> There is a pending a PR #10638 that will allow to warmup the tenant / shell to not have to wait for a first request.
> 
> That said, on a prod server if your app is recycled the bg task is stopped, here you may need a way to keep it alive.

What should i do for that right now for the quick fix?? please advise.

**Reply by zl2fxy** [reply]
Just request for the tenant's page
